core/util.go

```diff
@@ -422,3 +423,14 @@ func RetryBackoff(retries int, base, max time.Duration, factor float64) time.Dur
 	backoff *= (1 - retryJitter) + 2*retryJitter*mrand.Float64()
 	return time.Duration(backoff)
 }
+
+// HashNames returns the sha256 hash of the lowercased, comma joined
```

Comment:
Along with the feedback to pass the list of names to SA instead of a pre-hashed set, I'd recommend moving this HashNames function out of util and into SA.